# **INDIAN STARTUP ECOSYSTEM**

## Project Description
We embark on a journey of discovery as we leverage our data analysis expertise to uncover the untapped potential within the Indian startup ecosystem. This project is designed to not only decode the numbers but to distill insights that will guide our team towards a successful foray into this dynamic market.

## Scope of Work

- Conduct a thorough exploration of datasets, dissecting funding patterns, sectoral nuances, and geographical hotspots in the Indian startup landscap
- Analyze funding received by startups in india from 2018 to 2021



## Hypothesis 
**Null Hypothesis (H₀)**: There is no significant difference in the average funding received by startups and Location.

**Alternative Hypothesis (H₁)**: There are significant differences in the average funding received by startups and Location.

**Null Hypothesis (H0)**: There is no significant relationship between funding and the sector  

**Alternative Hypothesis (H1)**: There is a significant relationship between funding and the sector

## Questions 
1. How does funding vary across different industry sectors in India?
2. How does funding vary with the loaction of the start-ups
3. What is the relationship between the amount of funding and the stage of the company?
4. How have funding trends evolved between 2018 and 2021?
5. What are the most attractive sectors for investors?
6. Does the location of the company influence its sector?




# **DATA EXPLORATION, DATA UNDERSTANDING and DATA ANALYSIS**

In [1]:
# Load libraries
# Database connnection
import pyodbc     
from dotenv import dotenv_values    #import the dotenv_values function from the dotenv package

# Data manipulation
import pandas as pd
import numpy as np

# Data visualization
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

# filter warnings
import warnings 
warnings.filterwarnings('ignore')

## **1. Loading and Inspection of Data**

**Load data from the SQL server**

In [2]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')

# Get the values for the credentials you set in the '.env' file
server = environment_variables.get("SERVER")
database = environment_variables.get("DATABASE")
username = environment_variables.get("USERNAME")
password = environment_variables.get("PASSWORD")

In [3]:
# Create a connection string

connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"
    
connection = pyodbc.connect(connection_string)



In [4]:
# sql query to get 2020 data. 
query_2020="SELECT * FROM dbo.LP1_startup_funding2020"

# sql query to get 2021 data. 
query_2021="SELECT * FROM dbo.LP1_startup_funding2021"

In [5]:
    # load 2021 data
data_2021=pd.read_sql(query_2021,connection)

    # load 2020 data
data_2020=pd.read_sql(query_2020,connection)

**Load CSV Files**

In [6]:
# load 2019 data
data_2019=pd.read_csv(r'C:\Users\iamde\OneDrive\Desktop\jupyter\india_startup_data\startup_funding2019.csv')

    # load 2018 data
data_2018=pd.read_csv(r'C:\Users\iamde\OneDrive\Desktop\jupyter\india_startup_data\startup_funding2018.csv')


### **2. Data Exploration and Understanding**

**Preview Each dataset**

In [7]:
# preview the rows and columns for the 2018 dataset
data_2018.sample(5)

Company Name                                      Industry Round/Series  \
414         NIRA  Credit, Financial Services, FinTech, Lending         Seed   
449     Cardekho  Automotive, E-Commerce, Information Services     Series C   
83       Carmesi                                   Health Care         Seed   
188      Eduvanz                   Finance, Financial Services        Angel   
371     Sportido                          Social Media, Sports         Seed   

          Amount                     Location  \
414      1000000  Bangalore, Karnataka, India   
449     75000000      Gurgaon, Haryana, India   
83   ₹34,000,000      Gurgaon, Haryana, India   
188       500000   Mumbai, Maharashtra, India   
371            —  Noida, Uttar Pradesh, India   

                                         About Company  
414  NIRA promotes financial inclusion by creating ...  
449  Cardekho is a car search venture that helps us...  
83   Carmesi is a brand of high-end menstrual-hygie...  
188  Eduvanz is new, innovative finance company, wh...  
371  Sportido is a sports discovery app that helps ...

In [8]:
# get a sample of 2019 dataset
data_2019.sample(5)

Company/Brand  Founded HeadQuarter            Sector  \
28                     HungerBox   2015.0   Bangalore          Foodtech   
64                       Moms Co      NaN   New Delhi        E-commerce   
39  Lil’ Goodness and sCool meal      NaN   Bangalore  Food & Nutrition   
77                         Sarva      NaN     Chennai   Yoga & wellness   
38                   Freshokartz   2016.0      Jaipur     E-marketplace   

                                         What it does  \
28  Offers B2B tech platform to corporates, aimed ...   
64  It is into mother and baby care-focused consum...   
39  It empowers moms to provide the best of foods ...   
77  It drives a cultural change in making yoga-bas...   
38      Online fruits and vegetables delivery company   

                       Founders  \
28  Sandipan Mitra, Uttam Kumar   
64                Malika Sadani   
39                          NaN   
77               Sarvesh Shashi   
38                Rajendra Lora   

                                             Investor    Amount($)  \
28  Paytm, Asian VC fund NPTK, Sabre Partners, Neo...  $12,000,000   
64               Saama Capital, DSG Consumer Partners   $5,000,000   
39                      Sajan Pillai, Muthu Krishnan,     $400,000   
77                   Venture fund, Fireside Ventures.   $2,800,000   
38                                           ThinkLab     $150,000   

           Stage  
28      Series C  
64      Series B  
39           NaN  
77           NaN  
38  Pre series A

In [9]:
# get a sample of 2020 dataset
data_2020.sample(5)

Company_Brand  Founded HeadQuarter         Sector  \
934                      FirstCry   2010.0        Pune       E-market   
769                      NoBroker   2014.0   Bangalore    Real Estate   
280  Reliance Retail Ventures Ltd   2006.0      Mumbai         Retail   
479                      Fanspole   2015.0       Thane  Entertainment   
118                          Plum   2019.0   Bangalore     HealthTech   

                                          What_it_does  \
934                          Baby products marketplace   
769                  Online real estate rental startup   
280  Reliance Retail offers online shopping and a c...   
479        India's only social fantasy gaming platform   
118  High quality health insurance and health benefits   

                                          Founders  \
934                 Supam Maheshwari, Amitava Saha   
769  Amit Kumar Agarwal, Akhil Gupta, Saurabh Garg   
280                                  Mukesh Ambani   
479                Aniket Shinde, Hardik Gondaliya   
118                 Abhishek Poddar, Saurabh Arora   

                                     Investor        Amount     Stage column10  
934                    Softbank’s Vision Fund  2.960000e+08  Series E     None  
769                          General Atlantic  3.000000e+07      None     None  
280  Silver Lake, Mubadala Investment Company  7.000000e+10      None     None  
479               Mohan Verma, Better Capital           NaN  Pre-Seed     None  
118           Surge, Tanglin Venture Partners  4.100000e+06      None     None

In [10]:
# get a sample of 2021 dataset
data_2021.sample(5)

Company_Brand  Founded HeadQuarter                    Sector  \
626      Greenjoules   2018.0        Pune          Renewable Energy   
281  The Gourmet Jar   2012.0       Noida          Food & Beverages   
261      Practically   2018.0   Hyderabad                E-learning   
209         StackBOX   2016.0   Bangalore  Logistics & Supply Chain   
352        SarvaGram   2019.0      Mumbai                   FinTech   

                                          What_it_does  \
626  Greenjoules is a green energy technology start...   
281  The Gourmet Jar offers a range of premium cond...   
261  An experiential learning app for grades 6th to...   
209                            DISTRIBUTION SIMPLIFIED   
352  SarvaGram provides livelihood upgradation for ...   

                                              Founders  \
626  R. Sethunath, V S Shridhar, V. Radhika, Virara...   
281                                       Apeksha Jain   
261  Subbarao Siddabattula, Charu Noheria, Ilangove...   
209          Shanmukha Boora, Sabyasachi Bhattacharjee   
352                         Utpal Isser, Sameer Mishra   

                            Investor       Amount     Stage  
626               Blue Ashva Capital     $4500000  Series A  
281                DevX Venture Fund      $150000      None  
261  NB Ventures, Earlsfield Capital     $5000000      None  
209    Ecosystem Ventures, CMM group     $1200000      None  
352                Elevation Capital  $10,000,000  Series B

**Shape of the data**

In [11]:
# get the number of rows and columns for the datasets
print(f"The 2018 dataset has {data_2018.shape[0]} rows and {data_2018.shape[1]} Columns\n")
print(f"The 2019 dataset has {data_2019.shape[0]} rows and {data_2019.shape[1]} Columns\n")
print(f"The 2020 dataset has {data_2020.shape[0]} rows and {data_2020.shape[1]} Columns\n")
print(f"The 2021 dataset has {data_2021.shape[0]} rows and {data_2021.shape[1]} Columns\n\n")

The 2018 dataset has 526 rows and 6 Columns

The 2019 dataset has 89 rows and 9 Columns

The 2020 dataset has 1055 rows and 10 Columns

The 2021 dataset has 1209 rows and 9 Columns




**Info of the data**

In [12]:
# overview of 2018 dataset
data_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


In [13]:
# overview of 2019 dataset
data_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount($)      89 non-null     object 
 8   Stage          43 non-null     object 
dtypes: float64(1), object(8)
memory usage: 6.4+ KB


In [14]:
# overview of 2020 dataset
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.6+ KB


In [15]:
# overview of 2021 dataset
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   HeadQuarter    1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What_it_does   1209 non-null   object 
 5   Founders       1205 non-null   object 
 6   Investor       1147 non-null   object 
 7   Amount         1206 non-null   object 
 8   Stage          781 non-null    object 
dtypes: float64(1), object(8)
memory usage: 85.1+ KB


**Displaying datasets columns**

In [16]:
# 2021 data columns
data_2021.columns

Index(['Company_Brand', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage'],
      dtype='object')

In [17]:
# 2020 data columns

data_2020.columns

Index(['Company_Brand', 'Founded', 'HeadQuarter', 'Sector', 'What_it_does',
       'Founders', 'Investor', 'Amount', 'Stage', 'column10'],
      dtype='object')

In [18]:
# 2019 data columns

data_2019.columns

Index(['Company/Brand', 'Founded', 'HeadQuarter', 'Sector', 'What it does',
       'Founders', 'Investor', 'Amount($)', 'Stage'],
      dtype='object')

In [19]:
# 2018 data columns

data_2018.columns

Index(['Company Name', 'Industry', 'Round/Series', 'Amount', 'Location',
       'About Company'],
      dtype='object')

## **Observations:**
**Issues with the data**

1. There is a discrepancy in the naming conventions between the columns in the 2018 and 2019 datasets compared to the 2020 and 2021 datasets.

2. The 2018 dataset exhibits some missing columns, contributing to an incomplete representation of the data.

3. Conversely, the 2020 dataset contains an additional column that appears to be extraneous and does not serve a meaningful purpose in our analysis.

**Course of Action:**

1. **Missing Column Engineering for 2018:**
   - We will address the absence of certain columns in the 2018 dataset by employing data engineering techniques to create and populate the missing columns, ensuring a comprehensive and consistent dataset.

2. **Column Name Standardization:**
   - To establish uniformity and coherence across all datasets, we will embark on a column renaming process for the 2018 and 2019 datasets. This action aims to align the naming conventions with those observed in the 2020 and 2021 datasets, facilitating seamless data integration and analysis.

3. **Extraneous Column Removal in 2020:**
   - The redundant column identified in the 2020 dataset will be removed, streamlining the dataset and eliminating unnecessary elements that do not contribute to the overall analysis objectives.

These actions collectively enhance the integrity, consistency, and completeness of the dataset, paving the way for a more robust and coherent analytical process.
ical process.







# **3. Data Cleaning**

**Handling missing columns and currency signs in the 2018 dataset**

- The 2018 dataset has missing; 'founded', 'founders', and 'investor' columns


In [20]:
# Engineer missing columns for the 2018 dataset
columns_to_add = ['founded', 'founders', 'investor']
for column in columns_to_add:
    if column not in data_2018.columns:
        data_2018[column] = np.NaN

# Replace '₹', commas, '—', and "''" in 'Amount' column
data_2018['Amount'] = data_2018['Amount'].str.replace(',', '').str.replace('—', '').str.replace("''",'').replace('', np.nan)

# Conditionally apply multiplication only where '₹' is present
mask = data_2018['Amount'].str.contains('₹', na=False)
data_2018.loc[mask, 'Amount'] = data_2018.loc[mask, 'Amount'].str.replace('₹', '').astype(float) * 0.0146

**Data collection Year**

- There is need to add a column that represents the year each dataset was collected. This will help with handling the datasets after merging the dataframes 


In [21]:
# add year when data was collected column to every dataset
data_2018['data_year'] = pd.to_datetime('2018', format='%Y').year
data_2019['data_year'] = pd.to_datetime('2019', format='%Y').year
data_2020['data_year'] = pd.to_datetime('2020', format='%Y').year
data_2021['data_year'] = pd.to_datetime('2021', format='%Y').year

**Merge the dataframes**

**Notes**
- The function below concatenates the dataframes then renames the columns to ensure uniformity across the merged dataframe


In [47]:
# Define function to concatenate and rename columns
def concat_dfs(df_1, df_2, df_3, df_4):
    # Rename columns in individual DataFrames
    df_1.rename(columns={'Company Name': 'company_brand', 'Industry': 'sector', 'Round/Series': 'stage',
                         'Amount': 'amount($)', 'Location': 'location', 'About Company': 'about_company'},
                inplace=True)
    df_2.rename(columns={'Company/Brand': 'company_brand', 'Founded': 'founded', 'HeadQuarter': 'location',
                         'Sector': 'sector', 'What it does': 'about_company', 'Founders': 'founders',
                         'Investor': 'investor', 'Amount($)': 'amount($)', 'Stage': 'stage'},
                inplace=True)
    df_3.rename(columns={'Company_Brand': 'company_brand', 'Founded': 'founded', 'HeadQuarter': 'location',
                         'Sector': 'sector', 'What_it_does': 'about_company', 'Founders': 'founders',
                         'Investor': 'investor', 'Amount': 'amount($)', 'Stage': 'stage'},
                inplace=True)
    df_4.rename(columns={'Company_Brand': 'company_brand', 'Founded': 'founded', 'HeadQuarter': 'location',
                         'Sector': 'sector', 'What_it_does': 'about_company', 'Founders': 'founders',
                         'Investor': 'investor', 'Amount': 'amount($)', 'Stage': 'stage'},
                inplace=True)

    # Concatenate dataframes along the row axis and reset index
    df = pd.concat([df_1, df_2, df_3, df_4])

    return df


df = concat_dfs(data_2018, data_2019, data_2020, data_2021)


In [48]:
#Drop the extreneous column 10
df.drop('column10', axis=1, inplace= True)

**Cleaning 'Amount' column**

**Notes**  
- Remove all currency signs  

- Remove all other umwanted characters, words and symbols  

- Populate null values with the mode of the dataset  

- Convert the column from object to float

In [24]:

# Remove dollar sign
df['amount($)'] = df['amount($)'].replace('\$', '', regex=True)

# Remove commas
df['amount($)'] = df['amount($)'].str.replace(',', '')

# Remove all other irrelevant characters, words and symbols
df['amount($)'] = df['amount($)'].replace(["Upsparks", 'undisclosed', 'Undisclosed', "ah! Ventures", 
                                               "Pre-series A", "ITO Angel Network LetsVenture", 
                                               "JITO Angel Network LetsVenture", "Series C", 'Seed', ','], '')

# Convert the 'amount($)' column to numeric
df['amount($)'] = pd.to_numeric(df['amount($)'])

In [49]:
df['amount($)'].info()

<class 'pandas.core.series.Series'>
Index: 2879 entries, 0 to 1208
Series name: amount($)
Non-Null Count  Dtype 
--------------  ----- 
2474 non-null   object
dtypes: object(1)
memory usage: 45.0+ KB


**Cleaning data_year column**

**Notes**  


- Convert data type to period


In [27]:
# Convert the data_year column to date
df['data_year']=pd.to_datetime(df['data_year'], format='%Y')
df['data_year']=df['data_year'].dt.to_period('y')
# df['founded']=pd.to_datetime(df['founded']).dt.year

In [28]:
# check for nulls and duplicated
print(f"There are {df['data_year'].isna().sum()} Null values in the 'data_year' column")

There are 0 Null values in the 'data_year' column


In [29]:
df['data_year'].info()

<class 'pandas.core.series.Series'>
Index: 2879 entries, 0 to 1208
Series name: data_year
Non-Null Count  Dtype        
--------------  -----        
2879 non-null   period[Y-DEC]
dtypes: period[Y-DEC](1)
memory usage: 45.0 KB


**Cleaning 'founded' column**

**Notes**
- Handle nulls by populating with the 'bfill' method



In [30]:
print(f"There are {df['founded'].isna().sum()} Null values in the 'founded' column")

There are 769 Null values in the 'founded' column


**Notes**  
- There are 769 null values in the 'founded' column.  

- Since dropping the nulls will lead to a significant loss of our data, Backward fill will be used to fill the null values

In [51]:
# Convert to datetime
df['founded'] = pd.to_datetime(df['founded'], format='%Y')

# Convert to period
df['founded'] = df['founded'].dt.to_period('Y')


In [52]:
df['founded'].info()

<class 'pandas.core.series.Series'>
Index: 2879 entries, 0 to 1208
Series name: founded
Non-Null Count  Dtype        
--------------  -----        
2110 non-null   period[Y-DEC]
dtypes: period[Y-DEC](1)
memory usage: 45.0 KB


**Cleaning the 'founders' column**

In [35]:
# Remove unwanted characters
df['founders'] = df['founders'].replace(['...', np.nan], np.NaN)

# Check the number of NaN values in the 'founders' column
nan_count = df['founders'].isna().sum()

print(nan_count)

545


In [56]:
df['founders'].info()

<class 'pandas.core.series.Series'>
Index: 2879 entries, 0 to 1208
Series name: founders
Non-Null Count  Dtype 
--------------  ----- 
2334 non-null   object
dtypes: object(1)
memory usage: 45.0+ KB


**Cleaning **Stage** column**

Startups start with pre-seed, progress through seed, Series A, Series B, etc., securing resources for development and strategies. Additional rounds like Series C or D may follow. External funding at each stage fuels growth toward the venture's full potential.

**Pre-Seed Funding**  
Entrepreneurial idea in early development; small funds needed; limited informal channels for raising funds.

**Seed Funding**  
First official equity funding; investors provide funds for equity ownership.

**Series A Financing**  
First venture capital round; developed product, consistent revenue, long-term profit plan.

**Series B Financing**  
For established startups; substantial user base and revenue; funding for expansion.

**Series C and Beyond**  
Optional rounds for final push before IPO or unmet objectives; Series C is the third venture capital round.

**Initial Public Offering (IPO)**  
Process of offering corporate shares to the public; used for funding or divestment.

link: https://www.startupindia.gov.in/content/sih/en/funding.html

In [38]:
# Cleaning stage column
df['stage'].unique()
df['stage']=df['stage'].replace(['https://docs.google.com/spreadsheets/d/1x9ziNeaz6auNChIHnMI8U6kS7knTr3byy_YBGfQaoUA/edit#gid=1861303593',
'$6000000','$1000000','$300000','$1200000'],np.NaN)

In [39]:
# Standardize funding stages in the 'stage' column
df['stage'] = df['stage'].replace(['Series A', 'Seies A', 'Series A-1', 'Series A2', 'Series A+', 'Series A+'], 'Series A')
df['stage'] = df['stage'].replace(['Pre-seed', 'Pre-seed Round', 'Pre seed Round', 'Pre seed round'], 'Pre-Seed Stage')
df['stage'] = df['stage'].replace(['Pre series A', 'Pre-series A', 'Pre Series A', 'Pre series A1', 'Pre-series A1', 'Pre- series A'], 'Pre series A')
df['stage'] = df['stage'].replace(['Series B', 'Series B+', 'Series B2', 'Series B3'], 'Series B')
df['stage'] = df['stage'].replace(['Series C', 'Series C', 'Series C, D','Series C', 'Private Equity','PE', 'Post-IPO Equity','Series D', 'Series E', 'Series F', 'Series G', 'Series H', 'Series I','Series D1','Series F2', 'Series F1'], 'Series C and Beyond')
df['stage'] = df['stage'].replace(['Venture - Series Unknown', None,'Grant','Debt','Debt Financing','Post-IPO Debt','Non-equity Assistance','Bridge','Bridge Round','Fresh funding','Funding Round','Mid series','Edge',], 'unknown')
df['stage'] = df['stage'].replace(['Corporate Round','Undisclosed','Secondary Market','Pre-series','Post series A','Pre-series B','Pre-Series B','Pre series B','Pre-series C','Pre series C'], 'Other Stages')
df['stage'] = df['stage'].replace(['Seed','Seed funding','Pre-Seed','Angel', 'Angel Round','Seed fund', 'Seed round', 'Seed A','Seed Funding', 'Seed Round & Series A', 'Series E2', 'Seed Round','Seed Investment','Seed+','Early seed'],'Seed Stage')

In [57]:
df['stage'].info()

<class 'pandas.core.series.Series'>
Index: 2879 entries, 0 to 1208
Series name: stage
Non-Null Count  Dtype 
--------------  ----- 
1941 non-null   object
dtypes: object(1)
memory usage: 45.0+ KB


In [41]:
df['stage'].isna().sum()

0

In [42]:
df['stage'].value_counts()

stage
unknown                1060
Seed Stage              746
Series A                309
Pre series A            291
Series C and Beyond     232
Series B                138
Pre-Seed Stage           65
Other Stages             38
Name: count, dtype: int64

**Cleaning the Sector Column**

In [43]:
# Get the first sentence of every list
df['sector']=df['sector'].str.split(",").str[0]

In [44]:
# Get the first location from every list
df['location']=df['location'].str.split(",").str[0]

**Correct naming variations in the location column**

In [45]:
df['location']=df['location'].replace({'Bengaluru': 'Bangalore', 'Banglore': 'Bangalore', 'Gurugram': 'Gurgaon', 'Hyderebad': 'Hyderabad', 
                                       'New Delhi': 'Delhi', 'Ahmadabad': 'Ahmedabad', 'Ernakulam': 'Cochin', 'Telugana': 'Telangana',
                                         'Uttar pradesh': 'Uttar Pradesh', 'Rajastan': 'Rajasthan', 'San Franciscao': 'San Francisco', 
                                         'Samsitpur': 'Samastipur', 'Santra': 'Samtra', 'Rajsamand': 'Rajasthan', 'Kerala': 'Kochi', 
                                         'The Nilgiris': 'Nilgiris', 'Gurugram\t#REF!': 'Gurgaon', 'California': 'San Francisco', 
                                         'San Francisco Bay Area': 'San Francisco', 'Hyderebad': 'Hyderabad', 'Online Media\t#REF!': 'Online Media',
                                           'Pharmaceuticals\t#REF!': 'Pharmaceuticals', 'Information Technology & Services': 
                                           'IT Services', 'Small Towns': 'Unknown', 'Odisha': 'Odisha', 'Beijing': 'Beijing', 'Orissia': 'Odisha', 
                                           'Santra': 'Samtra', 'Vadodara': 'Vadodara', 'Food & Beverages': 'Food and Beverages', 'Bhilwara': 'Bhilwara',
                                             'Gandhinagar': 'Gandhinagar', 'Thiruvananthapuram': 'Thiruvananthapuram', 'Gurgaon': 'Gurgaon',
                                               'Patna': 'Patna', 'San Ramon': 'San Ramon', 'Plano': 'Plano', 'Bangaldesh': 'Bangladesh', 'Milano': 'Milano',
                                                 'California': 'California', 'Jharkhand': 'Jharkhand'}) 


In [46]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2879 entries, 0 to 1208
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype        
---  ------         --------------  -----        
 0   company_brand  2879 non-null   object       
 1   sector         2861 non-null   object       
 2   stage          2879 non-null   object       
 3   amount($)      1367 non-null   float64      
 4   location       2765 non-null   object       
 5   about_company  2879 non-null   object       
 6   founded        2110 non-null   period[Y-DEC]
 7   founders       2334 non-null   object       
 8   investor       2253 non-null   object       
 9   data_year      2879 non-null   period[Y-DEC]
dtypes: float64(1), object(7), period[Y-DEC](2)
memory usage: 247.4+ KB
